### PR TITLE
agent: unseal sealed secret env vars during exec requests

### LINKED ISSUE #13154: coco: Sealed Secrets not unsealed for exec requests
Sealed secrets stored in environment variables are unsealed when the container is created, but they are not unsealed when making an exec request. In other words, if you have a sealed secret attached to your container as an env var, the container entrypoint will see the unsealed environment variable, but if you exec into the container, you will see the sealed secret. For sealed secret volumes the unsealed secret will be available on the fs in either case.

You could argue that this is good, since it blocks exposing sealed secrets to non-secured exec requests. On the other hand, this is pretty confusing, and I think at least one person (besides myself) has been confused by it.

Should we add the secret resolving codepath to the exec request handler? Or we could just make a note about this in the docs.

cc @Xynnn007 @pawelpros

### PR DESCRIPTION
**Pull request by esposem**
Previously, sealed secret env vars were only unsealed during container creation (do_create_container via cdh_handler_sealed_secrets), but not during exec requests (do_exec_process). This meant that `kubectl exec` would show raw sealed plaintext for environment variables injected via secretKeyRef.

Call unseal_env on each env var in do_exec_process, gated on CDH client initialization, matching the existing cdh_handler_sealed_secrets. Extend tests to cover mixed sealed/plain env var lists and edge cases.

Fixes: #13154

### CODE REVIEW

**Comment by esposem** [comment]
CC @fitzthum @Xynnn007 @pawelpros (you're welcome Tobin)

File: src/agent/src/rpc.rs [hunk 1]
```diff
@@ -510,6 +510,18 @@ impl AgentService {
...
+        // Unseal any sealed secret env vars (mirrors cdh_handler_sealed_secrets)
+        if confidential_data_hub::is_cdh_client_initialized() {
+            for env in process.Env.iter_mut() {
+                match confidential_data_hub::unseal_env(env).await {
+                    Ok(unsealed_env) => *env = unsealed_env,
+                    Err(e) => {
+                        warn!(sl(), "Failed to unseal secret in exec: {}", e);
+                    }
+                }
+            }
```
Comment (stevenhorsman) [inline src/agent/src/rpc.rs]: This looks like a duplicate of the code in `cdh_handler_sealed_secrets`, so could we extract it out into an `unseal_envs(envs: &mut [String])` function?

**Review by stevenhorsman — COMMENTED** [review]
Refactor suggestion...

**Comment by sprt** [comment]
> Previously, sealed secret env vars were only unsealed during container creation (do_create_container via cdh_handler_sealed_secrets), but not during exec requests (do_exec_process). This meant that `kubectl exec` would show raw sealed plaintext for environment variables injected via secretKeyRef.

Does this mean that it's possible to call `kubectl exec` without `do_create_container` having been called?

**Comment by esposem** [comment]
> > Previously, sealed secret env vars were only unsealed during container creation (do_create_container via cdh_handler_sealed_secrets), but not during exec requests (do_exec_process). This meant that `kubectl exec` would show raw sealed plaintext for environment variables injected via secretKeyRef.
> 
> Does this mean that it's possible to call `kubectl exec` without `do_create_container` having been called?

No, `do_exec_process` always requires an existing container (it calls `get_container` and fails if not found). The issue is that each code path receives its own env vars independently: `do_create_container` gets them from the OCI spec and unseals them via `cdh_handler_sealed_secrets`, while `do_exec_process` gets a fresh set from the `ExecProcessRequest`. The exec request's env vars come directly from the kubelet/shim and are never passed through the OCI spec's unsealing path, so they arrive still sealed.

**Review by Xynnn007 — APPROVED** [review]
Thanks. This follows what we have discussed in previous thread.

File: src/agent/src/rpc.rs [hunk 2]
```diff
@@ -2698,6 +2702,18 @@ pub(crate) async fn cdh_secure_mount(
     Ok(())
 }
 
+async fn unseal_envs(envs: &mut [String]) -> Result<()> {
```
Comment (stevenhorsman) [inline src/agent/src/rpc.rs]: Given that this always returns ok, should we remove the return value if it's infallible?

**Review by stevenhorsman — COMMENTED** [review]
One nit in the new code

**Review by stevenhorsman — APPROVED** [review]

**Review by gkurz — APPROVED** [review]
/lgtm

Thanks @esposem !

**Comment by esposem** [comment]
All CI failures seem unrelated to the PR
